controllers/operatorpolicy_controller.go

```diff
@@ -1029,10 +1345,90 @@ func (r *OperatorPolicyReconciler) handleDeployment(
 	return updateStatus(policy, buildDeploymentCond(depNum > 0, unavailableDeployments), relatedObjects...), nil
 }
 
+func (r *OperatorPolicyReconciler) handleCRDs(
+	ctx context.Context,
+	policy *policyv1beta1.OperatorPolicy,
+	sub *operatorv1alpha1.Subscription,
+) ([]metav1.Condition, bool, error) {
+	if sub == nil {
+		return nil, updateStatus(policy, noCRDCond, noExistingCRDObj), nil
+	}
+
+	watcher := opPolIdentifier(policy.Namespace, policy.Name)
+	selector := subLabelSelector(sub)
+
+	crdList, err := r.DynamicWatcher.List(watcher, customResourceDefinitionGVK, sub.Namespace, selector)
+	if err != nil {
+		return nil, false, fmt.Errorf("error listing CRDs: %w", err)
+	}
+
+	// Same condition for musthave and mustnothave
+	if len(crdList) == 0 {
+		return nil, updateStatus(policy, noCRDCond, noExistingCRDObj), nil
+	}
+
+	relatedCRDs := make([]policyv1.RelatedObject, len(crdList))
+
+	if policy.Spec.ComplianceType.IsMustHave() {
+		for i := range crdList {
+			relatedCRDs[i] = matchedObj(&crdList[i])
```

Comment:
Nit: I believe this is the most efficient way since it avoids initializing policyv1.RelatedObject objects in the slice that go unused. This suggestion just sets the capacity of the underlying array. 
```suggestion
	relatedCRDs := make([]policyv1.RelatedObject, 0, len(crdList))

	if policy.Spec.ComplianceType.IsMustHave() {
		for i := range crdList {
			relatedCRDs = append(relatedCRDs, matchedObj(&crdList[i]))
```

Reply:
👍 Using `append` is probably better in general in case anything would ever exclude items in the future. Unlikely in this spot, but a good habit.